### PR TITLE
#164458127 A user should be able to delete a red-flag record

### PR DESCRIPTION
**Pull request by Siffersari**
### What does this PR do?
Adds an endpoint that deletes red-flag records

### Description of the task to be completed?
- Build out endpoint responsible for deleting red-flag records 

### How should you manually test this?

- [x] **Clone the repository**

        $ git clone https://github.com/ann-mukundi/ireporter.git

- [x] **Switch to testing branch**

       $ git checkout ft-user-delete-redflag-164458127 

- [x] **Switch to the ireporter directory**

       $ cd ireporter/

 

- [x] **Create and activate the virtual environment**

      $  virtualenv -p python .venv

      $ source .venv/bin/activate

- [x] **Install project dependencies**

       $ pip install -r api/requirements.txt

- [x] **Run the database migrations**

       $ python api/manage.py migrate

### Running the application

      $ python api/manage.py runserver

- [x] **Signup and activate user**

           POST <your-localhost>/api/auth/signup/

           POST <your-localhost>/api/auth/activate/

- [x] **Login user**

       POST <your-localhost>/api/auth/login/ 

       Use the token provided after login for Authorization and the endpoint
  
- [x] **Create a redflag record**

     

      POST <your-localhost>/api/redflags/ 

- [x] **Delete redflag record**

       DELETE <your-localhost>/api/redflags/int:redflag-id/ 

### Testing

      $ python api/manage.py test

### Prerequisites

Python, Git,  Virtualenv

### What are the relevant PT stories?

[#164458127](https://www.pivotaltracker.com/story/show/164458127)

### Screenshots
**Success**
Screenshot from 2019-03-14 01-42-19

**Non Existent Record**
Screenshot from 2019-03-14 01-42-42

**Validation**
Screenshot from 2019-03-14 01-40-42

### CODE REVIEW

**Comment by ElishaMisoi** [comment]
Good job. This looks good to me 👍🏽

**Review by Benkimeric — APPROVED** [review]
Good job @Siffersari. Personally, I approve this work

File: api/incidents/views.py [hunk 1]
```diff
@@ -54,3 +56,26 @@ def create_redflag_record(self, request):
     def create(self, request):
         request.data['createdBy'] = request.user.id
         return self.validate_record(request)
+
+    def destroy(self, request, **kwargs):
```
Comment (meshack-mbuvi) [inline api/incidents/views.py:60]: The name of this method is not very descriptive. You can give it a better name like `delete`

File: api/test/data/test_redflags.py [hunk 2]
```diff
@@ -92,6 +135,14 @@ def activate_account_and_login(self, activate_data=''):
 
         return self.response.data['token']
 
+    def delete_redflag(self, url='/api/redflags/int:redflag-id/'):
```
Comment (meshack-mbuvi) [inline api/test/data/test_redflags.py]: You should not specify the type(`int`) to the red-flag -d

**Review by meshack-mbuvi — CHANGES_REQUESTED** [review]

File: api/incidents/views.py (hunk 1 above)
Comment (Siffersari) [reply api/incidents/views.py:60]: I am using the destroy word to override the pre-existing method. I would love to request that it remains as to destroy for this approach to work.

**Comment by katherine95** [comment]
Great job.

File: api/test/data/test_redflags.py (hunk 2 above)
Comment (Siffersari) [reply api/test/data/test_redflags.py]: I have implemented this requested change

File: api/incidents/views.py (hunk 1 above)
Comment (meshack-mbuvi) [reply api/incidents/views.py:60]: Ok

**Review by AlvinMugambi — APPROVED** [review]
Good work

**Pull request merged**